src/store/account/utils.test.ts

```diff
@@ -317,4 +323,144 @@ describe('utils', () => {
 			);
 		});
 	});
+
+	describe('updateIdentities', () => {
+		it('should return undefined if the account field in the state is undefined', () => {
+			setupAccountStore();
+			useAccountStore.setState((state) => ({
+				...state,
+				account: undefined
+			}));
+			const state = useAccountStore.getState();
+
+			expect(updateIdentities(state, {}, [])).toEqual(undefined);
+		});
+
+		it('should return the original identities if no mods are passed', () => {
+			setupAccountStore();
+			const state = useAccountStore.getState();
+			expect(updateIdentities(state, {}, [])).toEqual(state.account?.identities.identity);
+		});
+
+		it('should return the original identities and the new identities', () => {
+			setupAccountStore();
+			const state = useAccountStore.getState();
+			const newIdentity: Identity = {
+				_attrs: {},
+				id: faker.string.uuid()
+			};
+			expect(updateIdentities(state, {}, [newIdentity])).toEqual(
+				expect.arrayContaining([...(state.account?.identities.identity ?? []), newIdentity])
+			);
```

Comment:
This is hiding the fact that the identities have an order.
It might be worth it to check the order too.